### PR TITLE
reg-util.py

### PR DESCRIPTION
**Pull request by mzieg**
not ready for test

### CODE REVIEW

**Comment by mzieg** [comment]
Ready for testing when appropriately controlled firmware is available to flash.

File: generic/reg-util.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        if not self.init_usb():
+            return
+
+        self.init_table()
+        self.init_gui()
+
+    def init_usb(self):
+        self.dev = None
+        for dev in usb.core.find(find_all=True):
+            if dev.idVendor == 0x24aa and dev.idProduct == 0x4000:
```
Comment (samiebee43) [inline generic/reg-util.py:98]: You can also do `usb.core.find(idVendor=0x24aa, idProduct=0x4000)`

File: generic/reg-util.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                value <<= 8
+        else:
+            # ...if it turns out to be big-endian, easy fix :-)
+            for i in range(3, -1, -1):
+                value |= buf[i]
+                value <<= 8
+
+        return value
+
+    def make_addr_combobox(self, row, col):
```
Comment (samiebee43) [inline generic/reg-util.py:206]: move to before init_gui (caller)

File: generic/reg-util.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        self.bind('<Control-R>', self.read_callback)
+        self.bind('<Control-W>', self.write_callback)
+        self.bind('<Control-V>', self.write_value.set_focus)
+
+    ############################################################################
+    # event callbacks
+    ############################################################################
+
+    def write_callback(self):
+        """ user clicked the "write" button """
```
Comment (samiebee43) [inline generic/reg-util.py:136]: Would be nice to have a comment providing some context about self.write_addr. 
```
# self.write_addr is a tk.StringVar contained in a ttk.ComboBox
```

It's also helpful to write callbacks using the `sender` parameter

**Comment by samiebee43** [comment]
Since this is a standalone script, these "quality control" type comments aren't all that important. But 2/3 of them reflect patterns from throughout Enlighten/Wasatch.PY. Namely omission of sender params in callbacks and not using usb.core.find search arguments. Also it was nice to review a complete file this time instead of a diff.

Next step I will checkout and test shortcuts, then finally we want to test on device.

**Review by samiebee43 — APPROVED** [review]

**Comment by mzieg** [comment]
merging to minimize drift before test

**Pull request merged**